src/WebHandlers.cpp

```diff
@@ -203,38 +207,45 @@ void AsyncStaticWebHandler::handleRequest(AsyncWebServerRequest* request) {
     // set etag to lastmod timestamp if available, otherwise to size
     String etag;
     if (lw) {
-      setLastModified(gmtime(&lw));
+      setLastModified(lw);
 #if defined(TARGET_RP2040)
       // time_t == long long int
-      const size_t len = 1 + 8 * sizeof(time_t);
+      constexpr size_t len = 1 + 8 * sizeof(time_t);
       char buf[len];
-      char* ret = lltoa(lw, buf, len, 10);
+      char* ret = lltoa(lw + request->_tempFile.size(), buf, len, 10);
       etag = ret ? String(ret) : String(request->_tempFile.size());
 #else
-      etag = String(lw);
+      etag = lw + request->_tempFile.size();   // etag combines file size and lastmod timestamp
 #endif
     } else {
-      etag = String(request->_tempFile.size());
+      etag = request->_tempFile.size();
     }
-    if (_last_modified.length() && _last_modified == request->header(T_IMS)) {
-      request->_tempFile.close();
-      request->send(304); // Not modified
-    } else if (_cache_control.length() && request->hasHeader(T_INM) && request->header(T_INM).equals(etag)) {
```

Comment:
why `_cache_control.length()` was checked before ? you know the reason ?

Reply:
No idea... I've scratched my head on this too... can't find any reason for this

Reply:
same!